src/OpenTelemetry.Instrumentation.Runtime/description.md

```diff
@@ -0,0 +1,198 @@
+
+| Name                                              | Description                                      | Units | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|---------------------------------------------------|--------------------------------------------------|-------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**gc.allocated.bytes**     | Bytes allocated over the lifetime of the process | `By`  | ObservableCounter | `Int64`    |                  |                  |
+| process.runtime.dotnet.**gc.fragmentation.ratio** | GC fragmentation ratio                           | `1`   | ObservableGauge   | `Double`   |                  |                  |
+
+- [GC.GetTotalAllocatedBytes](https://docs.microsoft.com/dotnet/api/system.gc.gettotalallocatedbytes):
+  Gets a count of the bytes allocated over the lifetime of the process. The returned
+value does not include any native allocations. The value is an approximate count.
+
+- GC fragmentation ratio is calculated as:
+If `gcMemoryInfo.HeapSizeBytes != 0`,
+the value is
+`gcMemoryInfo.FragmentedBytes * 1.0d / gcMemoryInfo.HeapSizeBytes`,
+otherwise the value is `0`, where `var gcMemoryInfo = GC.GetGCMemoryInfo()`.
+
+  - [GCMemoryInfo.FragmentedBytes](https://docs.microsoft.com/dotnet/api/system.gcmemoryinfo.fragmentedbytes?view=netcore-3.1):
+  Gets the total fragmentation when the last garbage collection occurred.
+  - [GCMemoryInfo.HeapSizeBytes](https://docs.microsoft.com/dotnet/api/system.gcmemoryinfo.heapsizebytes?view=netcore-3.1#system-gcmemoryinfo-heapsizebytes):
+  Gets the total heap size when the last garbage collection occurred.
+
+### Additional GC metrics only available for NET6_0_OR_GREATER
+
+| Name                                    | Description        | Units | Instrument Type | Value Type | Attribute Key(s) | Attribute Values           |
+|-----------------------------------------|--------------------|-------|-----------------|------------|------------------|----------------------------|
+| process.runtime.dotnet.**gc.committed** | GC Committed Bytes | `By`  | ObservableGauge | `Int64`    |                  |                            |
+| process.runtime.dotnet.**gc.heapsize**  |                    | `By`  | ObservableGauge | `Int64`    | gen              | gen0, gen1, gen2, loh, poh |
+
+- [GCMemoryInfo.TotalCommittedBytes](https://docs.microsoft.com/dotnet/api/system.gcmemoryinfo.totalcommittedbytes?view=net-6.0#system-gcmemoryinfo-totalcommittedbytes):
+Gets the total committed bytes of the managed heap.
+
+- [GC.GetGCMemoryInfo().GenerationInfo[i].SizeAfterBytes](https://docs.microsoft.com/dotnet/api/system.gcgenerationinfo):
+Represents the size in bytes of a generation on exit of the GC reported in GCMemoryInfo.
+(The number of generations `i` is limited by [GC.MaxGeneration](https://docs.microsoft.com/dotnet/api/system.gc.maxgeneration))
```

Comment:
`GC.MaxGeneration` returned `2` when GC Count has values for gen0, gen1, gen2 and GC heap size has values for all five generations. I will remove mention of this property and use 3 for GC Count and `GC.GetGCMemoryInfo().GenerationInfo.Length` for heap sizes.

Reply:
Removed.